dask/array/linalg.py

```diff
@@ -48,14 +48,32 @@ def tsqr(data, name=None, compute_svd=False, _max_vchunk_size=None):
     Singular Value Decomposition.  It requires that the input array have a
     single column of blocks, each of which fit in memory.
 
-    If blocks are of size ``(n, k)`` then this algorithm has memory use that
-    scales as ``n**2 * k * nthreads``.
-
     Parameters
     ----------
     data: Array
     compute_svd: bool
         Whether to compute the SVD rather than the QR decomposition
+    use_recursion: bool
+        Whether to use recursion
+
+    Notes
+    ----------
+    If blocks are of size ``(n, k)`` then this algorithm has memory use that
+    scales as ``n**2 * k * nthreads``.
```

Comment:
Is this statement still true?  Actually, was it ever true?  I'm surprised that it makes no mention of the size of the full array

Reply:
That's true for a classic sequential algorithm.  Does that also hold for the parallel algorithm here?

Reply:
Basically it's bounded by that and the number of nodes in the "tree" (of QR computations). Because in each step, we keep the Q's for reassembly.

Reply:
Do you have a sense for the relationship on how memory use if affected by block size and number of blocks?  Or is the result of this PR that we are no longer as dependent on that?  Previously we were very sensitive to blocking (as you discovered).  I'm curious if this has changed entirely back to the O(nm) space constraint of the sequential algorithm or if there is still some dependency on blocking.

Reply:
The sequential algo was at risk due to the single core compute, which would occur when there are many blocks. Generally, there would have to be a 2-3 times the size of the matrix of memory to store it and the intermediates (the data in the tree nodes get exponentially smaller. It's an inverted tree with the leaves being the heaviest.